tools/depends/download-files.include

```diff
@@ -0,0 +1,33 @@
+#
+HASH_FOUND = yes
+ifneq ($(SHA512),)
+  HASH_TYPE = sha512
+  HASH_TOOL ?= $(SHA512SUM)
+else ifneq ($(SHA256),)
+  HASH_TYPE = sha256
+  HASH_TOOL ?= $(SHA256SUM)
+else
+  HASH_FOUND = no
+endif
+ifeq ($(NATIVE_OS), osx)
+  HASH_TOOL = $(SHASUM)
+endif
+
+HASH_SUM = $($(shell echo $(HASH_TYPE) | tr  '[:lower:]' '[:upper:]'))
+
+.PHONY: $(TARBALLS_LOCATION)/$(ARCHIVE).$(HASH_TYPE)
+all: $(TARBALLS_LOCATION)/$(ARCHIVE).$(HASH_TYPE)
+
+$(TARBALLS_LOCATION)/$(ARCHIVE):
+	cd $(TARBALLS_LOCATION); $(RETRIEVE_TOOL) $(RETRIEVE_TOOL_FLAGS) $(BASE_URL)/$(ARCHIVE)
+
+$(TARBALLS_LOCATION)/$(ARCHIVE).$(HASH_TYPE): $(TARBALLS_LOCATION)/$(ARCHIVE)
+ifeq ($(HASH_FOUND),no)
+	cd $(TARBALLS_LOCATION); $(SHA512SUM) $(ARCHIVE) > $(ARCHIVE).sha512
```

Comment:
The case of no hash causes some issues on macos. SHA512SUM is empty.
Maybe in the case of no HASH_FOUND, we still want to set HASH_TOOL (sha512sum be default, and in the osx case we already set HASH_TOOL so no change needed) and then change the dirrect call from SHA512SUM to HASH_TOOL